core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -1646,9 +1665,13 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
                         PUT_REF_TO_ADDR(frame_lp + addr_ret,
                                         GET_REF_FROM_ADDR(frame_lp + addr2));
                 }
+                if (*FRAME_REF(addr1)) {
+                    /* If original ref is i31ref, should not set frameref for
+                     * target cell */
+                    SET_FRAME_REF(addr_ret);
```

Comment:
Should also handle `SELECT_T` in classic interpreter, in which the two frame refs are not cleared. Should clear the second and reserve the first.

Reply:
done, thanks~